ostree-pulp.sh

```diff
@@ -0,0 +1,405 @@
+    else
+        sudo virsh undefine "${IMAGE_KEY}"
+    fi
+    # Remove qcow2 file.
+    sudo rm -f "$LIBVIRT_IMAGE_PATH"
+    # Remomve tmp dir.
+    sudo rm -rf "$TEMPDIR"
+    # Stop httpd
+    sudo systemctl disable httpd --now
+}
+
+# Test result checking
+check_result () {
+    greenprint "Checking for test result"
+    if [[ $RESULTS == 1 ]]; then
+        greenprint "💚 Success"
+    else
+        redprint "❌ Failed"
+        clean_up
+        exit 1
+    fi
+}
+
+##################################################
+##
+## Upload ostree commit to pulp test
+##
+##################################################
+# Setup pulp server
+greenprint "📄 Setup pulp server with one container"
+mkdir -p settings pulp_storage pgsql containers
+echo "CONTENT_ORIGIN='http://$(hostname):8080'
+ANSIBLE_API_HOSTNAME='http://$(hostname):8080'
+ANSIBLE_CONTENT_HOSTNAME='http://$(hostname):8080/pulp/content'
+CACHE_ENABLED=True" >> settings/settings.py
+sudo podman run --detach \
+            --publish 9090:80 \
+            --name pulp \
+            --volume "$(pwd)/settings":/etc/pulp:Z \
+            --volume "$(pwd)/pulp_storage":/var/lib/pulp:Z \
+            --volume "$(pwd)/pgsql":/var/lib/pgsql:Z \
+            --volume "$(pwd)/containers":/var/lib/containers:Z \
+            --device /dev/fuse \
+            quay.io/pulp/pulp:nightly
+
+# Wait until pulp service is fully functional
+sleep 120
+
+# Rotate pulp admin password
+greenprint "📄 Rotate pulp admin password"
+/usr/bin/expect <<-EOF
+spawn sudo podman exec -it pulp bash -c "pulpcore-manager reset-admin-password"
+expect {
+"*password" { send "${PULP_PASSWORD}\r"; exp_continue }
+"*again" { send "${PULP_PASSWORD}\r" }
+}
+expect eof
+EOF
+
+# Write a pulp config file.
+greenprint "📄 Prepare pulp config file"
+tee "$PULP_CONFIG_FILE" > /dev/null << EOF
+provider = "pulp.ostree"
+
+[settings]
+server_address = "${PULP_SERVER}"
+repository = "${PULP_REPO}"
+basepath = "${PULP_BASEPATH}"
+username = "${PULP_USERNAME}"
+password = "${PULP_PASSWORD}"
+EOF
+
+greenprint "📄 pulp config file:"
+cat "$PULP_CONFIG_FILE"
+
+# Write a blueprint for ostree image.
+tee "$BLUEPRINT_FILE" > /dev/null << EOF
+name = "ostree"
+description = "A base ostree image"
+version = "0.0.1"
+modules = []
+groups = []
+
+[[packages]]
+name = "python3"
+version = "*"
+
+[[packages]]
+name = "sssd"
+version = "*"
+
+[[customizations.user]]
+name = "admin"
+description = "Administrator account"
+password = "\$6\$GRmb7S0p8vsYmXzH\$o0E020S.9JQGaHkszoog4ha4AQVs3sk8q0DvLjSMxoxHBKnB2FBXGQ/OkwZQfW/76ktHd0NX5nls2LPxPuUdl."
+key = "${SSH_KEY_PUB}"
+home = "/home/admin/"
+groups = ["wheel"]
+EOF
+
+greenprint "📄 ostree blueprint"
+cat "$BLUEPRINT_FILE"
+
+# Prepare the blueprint for the compose.
+greenprint "📋 Preparing blueprint"
+sudo composer-cli blueprints push "$BLUEPRINT_FILE"
+sudo composer-cli blueprints depsolve ostree
+
+# Build commit image
+build_image ostree "$IMAGE_TYPE" test "$PULP_CONFIG_FILE"
```

Comment:
What does `test` mean? If it's image_key, you have IMAGE_KEY defined already.

Reply:
the parameter test has no effect on this command, added comments inline to explain.